index.js

```diff
@@ -373,62 +373,86 @@
   ];
 
   //  ValidDate :: Type
-  $.ValidDate = $.NullaryType(
+  $.ValidDate = NullaryType(
     'sanctuary-def/ValidDate',
     function(x) { return $.Date.test(x) && !isNaN(x.valueOf()); }
   );
 
+  //  PositiveNumber :: Type
+  $.PositiveNumber = NullaryType(
+    'sanctuary-def/PositiveNumber',
+    function(x) { return $.Number.test(x) && x > 0; }
```

Comment:
> Why the inconsistency?

Because I don't want to shadow [`Number`][1]. I completely agree that this is inconsistent, but I don't currently see an elegant alternative.

> And, for that matter, why are these types not included in `$.env`?

The more types the environment contains, the more types may be listed in error messages:

```javascript
$.create($.env)('f', {}, [a, a], x => 'abc')(42);
// ! TypeError: ‘f’ is expected to return a value of type Number; returned "abc"

$.create($.env.concat([$.PositiveNumber, $.ValidNumber, $.NonZeroValidNumber, $.FiniteNumber, $.PositiveFiniteNumber, $.NonZeroFiniteNumber, $.Integer, $.PositiveInteger, $.NonZeroInteger]))('f', {}, [a, a], x => 'abc')(42);
// ! TypeError: ‘f’ is expected to return a value of type Number or PositiveNumber or ValidNumber or NonZeroValidNumber or FiniteNumber or PositiveFiniteNumber or NonZeroFiniteNumber or Integer or PositiveInteger or NonZeroInteger; returned "abc"
```

Since the number of types we could define is infinite, we must decide which types "deserve" to be included in `$.env`. Restricting `$.env` to built-ins provides clear delineation.


[1]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number